nixos/modules/services/audio/mpd.nix

```diff
@@ -123,10 +133,24 @@ in {
 
   config = mkIf cfg.enable {
 
+    systemd.sockets.mpd = mkIf cfg.startWhenNeeded {
+      description = "Music Player Daemon Socket";
+      wantedBy = [ "sockets.target" ];
+      listenStreams = [
+        "/run/mpd/socket"
+        "${cfg.network.listenAddress}:${toString cfg.network.port}"
+      ];
```

Comment:
You are right. I have removed the IPC socket for now. I couldn't find a client where the default is set to a local socket anyway. In the future we could add an option which allows to specify a path for it but I guess most people never even thought about using one.